pkg/infra/pulumi_aws/iac/api_gateway.ts

```diff
@@ -40,105 +42,154 @@ export class ApiGateway {
+        const resourceId = pulumi.interpolate`${api.id}/*/${verb == 'ANY' ? '*' : verb}${
+            path == '' ? '/' : path
+        }`
+        const lambda = this.lib.execUnitToFunctions.get(r.execUnitName)!
+        return new aws.lambda.Permission(
+            `${gwName}-${verb}-${path}`,
             {
-                apiId: api.id,
-                integrationType: 'AWS_PROXY',
-                connectionType: createVPC ? 'VPC_LINK' : 'INTERNET',
-                contentHandlingStrategy: 'CONVERT_TO_TEXT',
-                integrationMethod: verb,
-                integrationUri: invokeArn,
+                statementId: `${gwName}-${verb}-${r.path.replace(/[^a-zA-Z0-9]+/g, '-')}`,
+                action: 'lambda:InvokeFunction',
+                function: lambda,
+                principal: 'apigateway.amazonaws.com',
+                sourceArn: pulumi.interpolate`arn:aws:execute-api:${this.lib.region}:${this.accountId}:${resourceId}`,
             },
-            {
-                parent: api,
-            }
+            { parent: lambda }
         )
     }
 
-    createPrivateIntegration(
-        gwName: string,
-        api: aws.apigatewayv2.Api,
-        vpcLink: aws.apigatewayv2.VpcLink,
-        integrationUri: pulumi.Output<string>,
-        verb: string,
-        execUnitName: string
-    ): aws.apigatewayv2.Integration {
-        return new aws.apigatewayv2.Integration(
-            `${this.lib.name}-${gwName}-${execUnitName}`,
-            {
-                apiId: api.id,
-                description: 'Example with a load balancer',
-                integrationType: 'HTTP_PROXY',
-                integrationUri,
-                integrationMethod: verb,
-                connectionId: '${stageVariables.vpcLinkId}',
-                connectionType: 'VPC_LINK',
-                passthroughBehavior: 'WHEN_NO_MATCH',
-            },
-            {
-                dependsOn: [vpcLink],
-                parent: api,
-            }
-        )
+    integrationName(r: Route) {
+        const execUnit = this.lib.resourceIdToResource.get(`${r.execUnitName}_exec_unit`)
+        const verb = r.verb.toUpperCase()
+        const path = this.convertPath(r.path)
+        return `${verb}-${path}-${execUnit.type}`
+    }
+
+    createHTTPIntegration(api: aws.apigatewayv2.Api, r: Route): aws.apigatewayv2.Integration {
+        const execUnit = this.lib.resourceIdToResource.get(`${r.execUnitName}_exec_unit`)
+        const integrationName = this.integrationName(r)
+        switch (execUnit.type) {
+            case 'ecs':
+                const ecsNlb = this.lib.execUnitToNlb.get(r.execUnitName)!
+                return new aws.apigatewayv2.Integration(
+                    integrationName,
+                    {
+                        apiId: api.id,
+                        integrationType: 'HTTP_PROXY',
+                        integrationMethod: 'ANY',
+                        integrationUri: ecsNlb.loadBalancer.arn,
```

Comment:
this will only work if the nlb is an awsx nlb which i think we moved completely off of now

Reply:
nevermind

Reply:
i thought we migrated away from it but we didnt